tools/migration/internal/repo_migration_helper.go

```diff
@@ -0,0 +1,64 @@
+package internal
+
+import (
+	"os"
+	"strings"
+
+	"github.com/filecoin-project/go-filecoin/repo"
+)
+
+type RepoMigrationHelper struct {
+	oldRepoPath, newRepoPath, oldVersion, newVersion string
+}
+
+// NewRepoMigrationHelper takes options for old and new repo paths, figures out
+// what the correct paths should be, and creates a new RepoMigrationHelper with the
+// correct paths.
+func NewRepoMigrationHelper(oldRepoOpt, newRepoOpt, oldVersion, newVersion string) *RepoMigrationHelper {
+	oldPath := getOldRepoPath(oldRepoOpt)
+	return &RepoMigrationHelper{
+		newRepoPath: getNewRepoPath(oldPath, newRepoOpt, oldVersion, newVersion),
+		newVersion:  newVersion,
+		oldRepoPath: oldPath,
+		oldVersion:  oldVersion,
+	}
+}
+
+// GetOldRepo returns the old repo dir, opened as read-only.
+func (rmh *RepoMigrationHelper) GetOldRepo() (*os.File, error) {
+	return os.Open(getOldRepoPath(rmh.oldRepoPath))
+}
+
+// MakeNewRepo creates the new repo dir and returns it with Read/Write access.
+func (rmh *RepoMigrationHelper) MakeNewRepo() (*os.File, error) {
+	if err := os.Mkdir(rmh.newRepoPath, os.ModeDir|0744); err != nil {
```

Comment:
should this be 0700?

Reply:
I think it's fine.

Maybe be explicit that this will fail if such a directory already exists.